### ISSUE TITLE
@bsRecord's record being wrong isn't reported properly in bsb's output

### ISSUE THREAD

**Issue by justgage** [mhallin/graphql_ppx#42]
For instance this name should be a string, not an int

record:
![image](https://user-images.githubusercontent.com/164033/42061757-74f95fd0-7ae8-11e8-9589-0fd5e5aadc22.png)

graphql string
![image](https://user-images.githubusercontent.com/164033/42061878-dc8b82c2-7ae8-11e8-8876-b53cf4aeddb1.png)

How bsb reports it:
![image](https://user-images.githubusercontent.com/164033/42061748-67233f16-7ae8-11e8-9e5b-32170952138c.png)

It doesn't even tell me what file it's in, let alone the line number :/.

**Comment by thangngoc89** [comment]
@justgage try to run the bsb command separately in the terminal. Reason-scripts is eating the error report.

**Comment by mhallin** [comment]
What's the type signature for the `Id.fromString` function, and what does the `id` field return?

**Comment by justgage** [comment]
@thangngoc89 I would say it's still not very clear: 
![image](https://user-images.githubusercontent.com/164033/42099164-f97526a4-7b79-11e8-85bc-25b33d52c614.png)

@mhallin The `Id.fromString` is just `id => string` where id is an abstract type. However this isn't the part that's breaking in this example.

Maybe the real thing that's making this output unclear is the error pretty printer?

**Comment by mhallin** [comment]
Ah, sorry for the irrelevant question. I see the problem now, and yes that's a bug in the record decoder, I suppose. Thanks for reporting this!

**Comment by Anahkiasen** [comment]
Having the same issue on a much more basic level:

Data.re
```reason
type total = {
  month: string,
  amount: int,
};

type account = {
  id: int,
  name: string,
  overdraft: int,
  totals: array(total),
};

type accounts = array(account);
```

App.re
```reason
open Data;

module GetAccounts = [%graphql {|
    query FetchAccounts {
        accounts @bsRecord {
            id
            name
            overdraft
            totals {
                month
                amount
            }
        }
    }
|}];
```

With this schema:

```graphql
type Total {
    month: String!
    amount: Int!
}

type Account {
    id: ID!
    name: String!
    overdraft: Int!
    balance: Int!
    savings: Int
    expenditures: [Expenditure!]!
    totals: [Total!]!
}

type Query {
    accounts: [Account!]!
}
```

Gets me this:

```
[1/1] Building BsReactstrap.cmi
ninja: no work to do.
[5/10] Building src/App.mlast
[dirty_checker][update ok]
[10/10] Building src/App.mlast.d
[4/5] Building src/App.cmj

  We've found a bug for you!
  (No file name)

  This has type:
    string
  But somewhere wanted:
    int

error Command failed with exit code 2.
```

Is this the correct way to type an array of records? Or do I need to type all existing fields in the record type and not just the ones I use in the query?

**Comment by mhallin** [comment]
Yeah, every subselection will need an `@bsRecord` directive.

Another issue with your data model is that the `ID` type is decoded into a string, but your `account` type has the ID as an integer.

**Comment by mhallin** [comment]
The original issue - incorrect source location for record fields - is fixed in version 0.2.8.
